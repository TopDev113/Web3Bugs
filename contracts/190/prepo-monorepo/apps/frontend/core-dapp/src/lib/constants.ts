export const PROJECT_NAME = 'core-dapp'
export const PREPO_WEBSITE = 'https://prepo.io'
export const PREPO_TWITTER = 'https://twitter.com/prepo_io'
export const PREPO_DISCORD = 'https://url.prepo.io/discord-dapp'
export const PREPO_MEDIUM = 'https://medium.com/prepo'
export const PREPO_TESTNET_FORM = 'https://url.prepo.io/whitelist-dapp'

export const FEE_DENOMINATOR = 1000000
export const VALUATION_DENOMINATOR = 1000000
export const TWO_DECIMAL_DENOMINATOR = 100
export const DECIMAL_LIMIT = 0.01
export const ONE_DAY_UNIX = 24 * 60 * 60
export const ONE_HOUR_UNIX = 60 * 60
export const CURRENCY_PRECISION = 2

export const USDC_SYMBOL = 'FAKEUSD'
export const FAKEUSD_AIRDROPPED_ON_TESTNET = 100

// this will directly affect query data of market chart
// when querying historical data from subgraphs, if we're expecting large amount of data
// it's better to specify from what time onwards we want to query to reduce the number of data we're searching
export const PROJECT_START_TIMESTAMP = 1619170975

export const TRANSACTION_SETTING = {
  DEFAULT_AMOUNT: 1000,
  MIN_TRADE_AMOUNT: 0,
  MAX_TRADE_AMOUNT: 20000,
}

export enum BigNumberPadding {
  ONE = 10,
  TWO = 100,
  THREE = 1000,
  FOUR = 10000,
}

export const CURRENTPRICE_MULTIPLIER = BigNumberPadding.THREE
export const SLIPPAGE_MULTIPLIER = BigNumberPadding.THREE

export const ERC20_UNITS = 18

export const CHARACTERS_ON_WALLET = 42
