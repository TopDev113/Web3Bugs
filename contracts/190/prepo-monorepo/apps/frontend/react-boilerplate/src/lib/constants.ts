export const PROJECT_NAME = 'boilerplate'

export const APP_NAME = 'prePO'

export const ERC20_UNITS = 18
