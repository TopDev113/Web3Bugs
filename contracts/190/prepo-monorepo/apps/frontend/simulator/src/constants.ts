export const DELTA = 0.000001
export const FEE_STEP = 0.005
export const REWARD_STEP = 0.05
export const SWAP_FEE_APY_STEP = 0.01
export const MAX_FEE = 0.05
export const MAX_REWARD = 5
export const ONE_STEP = 0.01
export const MAX_DEPOSIT_AMOUNT = 10000000
export const UNKNOWN_TABLE_TOP_ROI = '$ ???'
export const UNKNOWN_TABLE_ROW_ROI = '???'
export const DEFAULT_HOLDING_PERIOD_MESSAGE = 'Calculated value assumes a 12 month holding period.'

export const STEP_NUMBER_INPUT_STEPS = {
  LIMIT_10: 0.1,
  FIVE_STEP: REWARD_STEP,
  LIMIT_1: 0.01,
  ONE_STEP,
  DECIMAL_STEP: 0.001,
}
