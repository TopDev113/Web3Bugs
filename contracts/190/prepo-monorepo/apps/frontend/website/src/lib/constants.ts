export const IS_BROWSER = Boolean(process.browser)

export const ONE_SECOND = 1000

export const PROJECT_NAME = 'website'

export const PRODUCTION_DOMAIN_EN = ''

export const DEFAULT_LANGUAGE = 'en-us'

export const APP_NAME = 'prePO'

export enum ROUTES {
  APP = 'https://app.prepo.io',
  BLOG = 'https://medium.com/prepo',
  DOCS = 'https://docs.prepo.io',
  JOBS = 'https://url.prepo.io/jobs',
  NEWSLETTER = 'https://url.prepo.io/newsletter-website',
  PRESALE_BLOG = 'https://url.prepo.io/blog-presale-website',
  PUBLIC_SALE_BLOG = 'https://medium.com/prepo/announcing-the-ppo-token-public-sale-6f672e23325a',
  TELEGRAM = 'https://t.me/prePO_News',
  GITHUB = 'https://github.com/prepo-io',
  SIMULATOR = 'https://simulator.prepo.io',
  TOKEN_SALE = 'https://sale.prepo.io/',
}

export const MAILCHIMP_URL =
  'https://prepo.us6.list-manage.com/subscribe/post?u=b8c56831dea49f16a5f4d0518&id=111d836609'

// https://stackoverflow.com/a/201378
export const EMAIL_REGEX =
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/

export const EMPTY_RECAPTCHA_SITE_ID = '__EMPTY_RECAPTCHA_SITE_ID'
export const RECAPTCHA_SITE_ID =
  process.env.NEXT_PUBLIC_RECAPTCHA_SITE_ID || '6Ld2IqcdAAAAAEogBZJGEJ_3s0vpsuHJ4UK0aECw'
export const MAILCHIMP_SUBSCRIBE_URL =
  process.env.NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_URL || 'https://mailchimp-subscribe.prepo.workers.dev'

export const PANELBEAR_SITE_ID = 'EZSAi3UGpKr'
