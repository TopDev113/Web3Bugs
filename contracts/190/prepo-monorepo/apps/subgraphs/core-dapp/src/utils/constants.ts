import { BigDecimal, BigInt } from '@graphprotocol/graph-ts'

// ... addresses ...
export const COLLATERAL_TOKEN_ADDRESS = '0xab7f09a1bd92ae0508884e6ab02a7a11df83512d'
export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'

// ... numbers ...
export const DEFAULT_LONG_SHORT_DECIMALS = 18
export const ONE_BI = BigInt.fromI32(1)
export const ZERO_BI = BigInt.fromI32(0)
export const ZERO_BD = BigDecimal.fromString('0')

// ... transactions ...
// wasm/assemblyscript can't compile objects nor enum
// so we must set these separately

export const EVENTS_TRANSFER = 'TRANSFER'
export const EVENTS_SWAP = 'SWAP'

export const ACTIONS_SEND = 'SEND'
export const ACTIONS_RECEIVE = 'RECEIVE'
export const ACTIONS_OPEN = 'OPEN'
export const ACTIONS_CLOSE = 'CLOSE'
export const ACTIONS_POOL = 'POOL'

// ... token types ...
export const TOKEN_TYPE_COLLATERAL = 'COLLATERAL'
export const TOKEN_TYPE_LONG_SHORT = 'LONG_SHORT'
export const TOKEN_TYPE_COLLATERAL_BASE = 'COLLATERAL_BASE'

export class HistoricalEventTypes {
  deposit: string = 'DEPOSIT'
  withdraw: string = 'WITHDRAW'
  open: string = 'OPEN'
  close: string = 'CLOSE'
}
