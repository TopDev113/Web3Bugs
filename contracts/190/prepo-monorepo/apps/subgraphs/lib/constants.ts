import { BigDecimal, BigInt } from '@graphprotocol/graph-ts'

// ... addresses ...
export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'

// ... numbers ...
export const DEFAULT_LONG_SHORT_DECIMALS = 18
export const ONE_BI = BigInt.fromI32(1)
export const ZERO_BI = BigInt.fromI32(0)
export const ZERO_BD = BigDecimal.fromString('0')
