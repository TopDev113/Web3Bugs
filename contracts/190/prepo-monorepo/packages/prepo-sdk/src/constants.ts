const DELTA = 0.00001
const INITIAL_LP_SUPPLY = 1000

const EMPTY_POOL_BALANCE = Object.freeze({ long: 0, short: 0 })

const EMPTY_MARKET_BALANCE = Object.freeze({
  long: 0,
  short: 0,
})

const SEC_IN_MS = 1000
const MIN_IN_MS = SEC_IN_MS * 60

export { DELTA, INITIAL_LP_SUPPLY, EMPTY_POOL_BALANCE, MIN_IN_MS, EMPTY_MARKET_BALANCE }
