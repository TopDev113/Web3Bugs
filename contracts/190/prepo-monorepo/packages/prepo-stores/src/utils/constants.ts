import { Network, NETWORKS, SupportedNetworks } from 'prepo-constants'

export const DYNAMIC_CONTRACT_ADDRESS = 'DYNAMIC'

export const getNetworkByChainId = (chainId: number | undefined): Network | undefined => {
  // eslint-disable-next-line no-restricted-syntax
  for (const key in NETWORKS) {
    if ({}.hasOwnProperty.call(NETWORKS, key)) {
      const network = key as SupportedNetworks
      if (NETWORKS[network].chainId === chainId) {
        return NETWORKS[network]
      }
    }
  }

  return undefined
}
