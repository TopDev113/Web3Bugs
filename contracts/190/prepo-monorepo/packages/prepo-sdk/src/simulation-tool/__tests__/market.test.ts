import Market from '../market'
import Actor from '../actor'
import { DELTA, EMPTY_MARKET_BALANCE } from '../../constants'
import Pool from '../pool'

const config = {
  bounds: { ceil: 0.8, floor: 0.2 },
  fee: 0.02,
  protocolFee: 0.01,
}
const initialDeposit = 1000
const marketName = 'test-market'
let market: Market
let creator: Actor

describe('a market is created', () => {
  beforeEach(() => {
    creator = new Actor('creator', initialDeposit)
    market = new Market(marketName, config, initialDeposit, creator)
  })

  test('the name is set correctly', () => {
    expect(market.name).toEqual(marketName)
  })

  test('the config is set correctly', () => {
    expect(market.config).toEqual(config)
  })

  test('long pool is initialized with correct balances', () => {
    expect(market.longPool.realReserves).toMatchSnapshot()
  })

  test('short pool is initialized with correct balances', () => {
    expect(market.shortPool.realReserves).toMatchSnapshot()
  })

  test('convertedStableDeposit is initialized correctly', () => {
    expect(market.convertedStableTokens).toMatchSnapshot()
  })

  test('creator has no leftover balance', () => {
    expect(market.getActorBalance(creator)).toEqual(EMPTY_MARKET_BALANCE)
  })

  test('creator usd balance is decremented correctly', () => {
    expect(creator.getUsdBalance()).toEqual(0)
  })

  test('actorStableIn is set correctly', () => {
    expect(market.getActorStableIn(creator)).toEqual(initialDeposit)
  })

  test('creator net worth set correctly', () => {
    const creatorNetWorth = market.getActorNetWorth(creator)
    expect(creatorNetWorth).toMatchSnapshot()
  })

  describe('a second LP deposits', () => {
    let secondLp: Actor
    const secondLpSize = 500
    beforeEach(() => {
      secondLp = new Actor('second-lp', secondLpSize)
      market.deposit(secondLp, secondLpSize)
    })

    test('convertedStableDeposit is updated correctly', () => {
      expect(market.convertedStableTokens).toMatchSnapshot()
    })

    test('secondLp has no leftover balance', () => {
      expect(market.getActorBalance(secondLp)).toEqual(EMPTY_MARKET_BALANCE)
    })

    test('secondLp usd balance is decremented correctly', () => {
      expect(secondLp.getUsdBalance()).toEqual(0)
    })

    test('actorStableIn is set correctly', () => {
      expect(market.getActorStableIn(secondLp)).toEqual(secondLpSize)
    })

    describe('a trader opens a long position', () => {
      let trader: Actor
      const initialPositionSize = 100
      beforeEach(() => {
        trader = new Actor('trader', initialPositionSize)
        market.openPosition(trader, 'long', initialPositionSize)
      })

      test('trader stable balance is updated correctly', () => {
        expect(trader.getUsdBalance()).toEqual(0)
      })

      test('trader long balance is updated correctly', () => {
        expect(market.getActorBalance(trader)).toMatchSnapshot()
      })

      test('trader net worth is set correctly', () => {
        const traderNetWorth = market.getActorNetWorth(trader)
        // slightly lower than their trade size due to arbitrager pushing price down
        expect(traderNetWorth).toMatchSnapshot()
      })

      test('trader net profit is set correctly', () => {
        const traderNetProfit = market.getActorNetProfit(trader)
        expect(traderNetProfit).toMatchSnapshot()
      })

      test('trader stable deposit is recorded correctly', () => {
        expect(market.getActorStableIn(trader)).toEqual(initialPositionSize)
      })

      test('the pools have been kept balanced by arbitragers', () => {
        const longPrice = Pool.calcMarketTokenPrice(market.longPool)
        const shortPrice = Pool.calcMarketTokenPrice(market.shortPool)
        const result = Math.abs(longPrice - shortPrice) < DELTA
        expect(result).toBe(false)
      })

      describe('the trader closes half their position', () => {
        beforeEach(() => {
          market.closePosition(trader, 'long', market.getActorBalance(trader).long / 2)
        })

        test('trader stable balance is updated correctly', () => {
          expect(trader.getUsdBalance()).toMatchSnapshot()
        })

        test('trader long balance is updated correctly', () => {
          expect(market.getActorBalance(trader)).toMatchSnapshot()
        })

        test('trader stable withdrawal is recorded correctly', () => {
          expect(market.getActorStableOut(trader)).toMatchSnapshot()
        })

        test('trader net worth is updated correctly', () => {
          const traderNetWorth = market.getActorNetWorth(trader)
          expect(traderNetWorth).toMatchSnapshot()
        })

        test('trader net profit is updated correctly', () => {
          const traderNetProfit = market.getActorNetProfit(trader)
          expect(traderNetProfit).toMatchSnapshot()
        })

        test('trader amountOut set correctly', () => {
          const amountOut = market.getActorStableOut(trader)
          expect(amountOut).toMatchSnapshot()
        })

        test('the pools have been kept balanced by arbitragers', () => {
          const longPrice = Pool.calcMarketTokenPrice(market.longPool)
          const shortPrice = Pool.calcMarketTokenPrice(market.shortPool)
          const result = Math.abs(longPrice - shortPrice) < DELTA
          expect(result).toBe(false)
        })
      })
    })

    describe('a trader opens a short position', () => {
      let trader: Actor
      const initialPositionSize = 300
      beforeEach(() => {
        trader = new Actor('trader', initialPositionSize)
        market.openPosition(trader, 'short', initialPositionSize)
      })

      test('trader stable balance is updated correctly', () => {
        expect(trader.getUsdBalance()).toEqual(0)
      })

      test('trader long balance is updated correctly', () => {
        expect(market.getActorBalance(trader)).toMatchSnapshot()
      })

      test('trader net worth is set correctly', () => {
        const traderNetWorth = market.getActorNetWorth(trader)
        // slightly lower than their trade size due to arbitrager pushing price down
        expect(traderNetWorth).toMatchSnapshot()
      })

      test('trader net profit is set correctly', () => {
        const traderNetProfit = market.getActorNetProfit(trader)
        expect(traderNetProfit).toMatchSnapshot()
      })

      test('trader stable deposit is recorded correctly', () => {
        expect(market.getActorStableIn(trader)).toEqual(initialPositionSize)
      })

      test('the pools have been kept balanced by arbitragers', () => {
        const longPrice = Pool.calcMarketTokenPrice(market.longPool)
        const shortPrice = Pool.calcMarketTokenPrice(market.shortPool)
        const result = Math.abs(longPrice - shortPrice) < DELTA
        expect(result).toBe(false)
      })

      describe('a third LP makes a deposit at bottom of range', () => {
        let thirdLp: Actor
        const positionSize = 200
        beforeEach(() => {
          thirdLp = new Actor('third-lp', positionSize)
          market.deposit(thirdLp, positionSize)
        })

        test('third LP should have 0 market token balances', () => {
          const bal = market.getActorBalance(thirdLp)
          const balanceLong = Math.abs(bal.long) < DELTA
          const balanceShort = Math.abs(bal.short) < DELTA
          expect(balanceLong).toBe(true)
          expect(balanceShort).toBe(true)
        })

        test('third LP net worth should be calculated correctly', () => {
          expect(market.getActorNetWorth(thirdLp)).toMatchSnapshot()
        })

        describe('trader trades price to top of the range', () => {
          beforeEach(() => {
            market.tradeToTargetPrice(trader, config.bounds.ceil)
          })

          test('third LP net worth should be calculated correctly', () => {
            expect(market.getActorNetWorth(thirdLp)).toMatchSnapshot()
          })
        })
      })
    })
  })
})
