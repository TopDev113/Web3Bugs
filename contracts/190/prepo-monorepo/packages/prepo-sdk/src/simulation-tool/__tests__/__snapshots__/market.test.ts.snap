// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader amountOut set correctly 1`] = `46.11469263897196`;

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader long balance is updated correctly 1`] = `
Object {
  "long": 91.11350587167246,
  "short": 0,
}
`;

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader net profit is updated correctly 1`] = `
Object {
  "amount": -6.674657608466674,
  "percent": -0.06674657608466675,
}
`;

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader net worth is updated correctly 1`] = `47.210649752561366`;

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader stable balance is updated correctly 1`] = `46.11469263897196`;

exports[`a market is created a second LP deposits a trader opens a long position the trader closes half their position trader stable withdrawal is recorded correctly 1`] = `46.11469263897196`;

exports[`a market is created a second LP deposits a trader opens a long position trader long balance is updated correctly 1`] = `
Object {
  "long": 182.22701174334492,
  "short": 0,
}
`;

exports[`a market is created a second LP deposits a trader opens a long position trader net profit is set correctly 1`] = `
Object {
  "amount": -2.512443379344674,
  "percent": -0.02512443379344674,
}
`;

exports[`a market is created a second LP deposits a trader opens a long position trader net worth is set correctly 1`] = `97.48755662065533`;

exports[`a market is created a second LP deposits a trader opens a short position a third LP makes a deposit at bottom of range third LP net worth should be calculated correctly 1`] = `199.347848471234`;

exports[`a market is created a second LP deposits a trader opens a short position a third LP makes a deposit at bottom of range trader trades price to top of the range third LP net worth should be calculated correctly 1`] = `170.45660184020414`;

exports[`a market is created a second LP deposits a trader opens a short position trader long balance is updated correctly 1`] = `
Object {
  "long": 0,
  "short": 478.63959455277376,
}
`;

exports[`a market is created a second LP deposits a trader opens a short position trader net profit is set correctly 1`] = `
Object {
  "amount": -17.040344099471668,
  "percent": -0.05680114699823889,
}
`;

exports[`a market is created a second LP deposits a trader opens a short position trader net worth is set correctly 1`] = `282.95965590052833`;

exports[`a market is created a second LP deposits convertedStableDeposit is updated correctly 1`] = `541.3136068524931`;

exports[`a market is created convertedStableDeposit is initialized correctly 1`] = `356.2164306640625`;

exports[`a market is created creator net worth set correctly 1`] = `990.4515486605293`;

exports[`a market is created long pool is initialized with correct balances 1`] = `
Object {
  "market": 356.2164306640625,
  "stable": 316.5177091247838,
}
`;

exports[`a market is created short pool is initialized with correct balances 1`] = `
Object {
  "market": 356.2164306640625,
  "stable": 316.5177091247837,
}
`;
